android/src/main/kotlin/co/sunnyapp/flutter_contact/resolver-extensions.kt

```diff
@@ -10,7 +10,8 @@ import android.provider.ContactsContract
 import androidx.annotation.RequiresApi
 import io.flutter.plugin.common.MethodChannel
 import java.io.ByteArrayOutputStream
-import java.time.OffsetDateTime
+import java.time.Instant
```

Comment:
Will it work? Isn't it Java 8?

Reply:
I tested it both on Android Nougat (7.0) which is API version 24 and Android KitKat (4.4) which is API version 19 and it worked. I no longer have an application crash, I can retrieve contacts.

Reply:
To complete my answer : 
"In many cases a user might need to ensure interoperability with third-party libraries that rely on the old date library.

In Java 8 old date library classes have been extended with methods that convert them to corresponding objects from new Date API.
New classes provide similar functionalities.

	
Instant instantFromCalendar = GregorianCalendar.getInstance().toInstant();
ZonedDateTime zonedDateTimeFromCalendar = new GregorianCalendar().toZonedDateTime();
Date dateFromInstant = Date.from(Instant.now());
GregorianCalendar calendarFromZonedDateTime = GregorianCalendar.from(ZonedDateTime.now());
Instant instantFromDate = new Date().toInstant();
ZoneId zoneIdFromTimeZone = TimeZone.getTimeZone("PST").toZoneId();"

Even if it is Java 8, it is just a way to map the new Java 8 way into the old Java 6 Date class. It fully works on any Java 6 device.